lib/buildreq.c

```diff
@@ -175,7 +175,7 @@ int rc_aaa_ctx_server(rc_handle * rh, RC_AAA_CTX ** ctx, SERVER * aaaserver,
 
 		result = rc_send_server_ctx(rh, ctx, &data, msg, type);
 
-		if ((result == OK_RC) || (result == CHALLENGE_RC)) {
```

Comment:
That would cause a memory leak on existing applications which receive a reject. An alternative approach would be to do that for applications that request that. That is you could introduce ```rc_set_flags()``` to set a flag such as ```RC_FLAG_AVPAIR_ON_REJECT```, and only then return that.

@martin-belanger you have added the EAP support, what do you think?